corehq/form_processor/submission_post.py

```diff
@@ -118,9 +118,8 @@ def _handle_basic_failure_modes(self):
 
     def _post_process_form(self, xform):
         self._set_submission_properties(xform)
-        if xform.is_submission_error_log:
-            found_old = scrub_meta(xform)
-            legacy_notification_assert(not found_old, 'Form with old metadata submitted', xform.form_id)
```

Comment:
Seems like this has been broken since we moved the code around. Not sure why it ended up under the `if`